airflow/jobs.py

```diff
@@ -675,10 +674,11 @@ def manage_slas(self, dag, session=None):
             qry = (
                 session
                 .query(TI)
-                .filter(TI.state != State.SUCCESS)
-                .filter(TI.execution_date.in_(sla_dates))
-                .filter(TI.dag_id == dag.dag_id)
-                .all()
+                .filter(
+                    TI.state != State.SUCCESS,
+                    TI.execution_date.in_(sla_dates),
+                    TI.dag_id == dag.dag_id
+                ).all()
```

Comment:
This change can bring minor performance improvement. 

Benchmarking for this sort of change was done in https://github.com/apache/airflow/pull/4433

Reply:
do we know the actual sql difference between these two approach?

Reply:
### SqlAlchemy

`Multiple criteria may be specified as comma separated; the effect is that they will be joined together using the and_() function`

Reference: https://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.filter

### SQL Statements Under the Hood
I have checked the sql statements below:
```python
from airflow.settings import Session
from airflow import models
s = Session()
DM = models.DagModel

print(s.query(DM).filter(DM.is_active).filter(DM.dag_id == 'tutorial').statement)
print(s.query(DM).filter(DM.is_active, DM.dag_id == 'tutorial').statement)
```

Result:
```
SELECT dag.dag_id, dag.is_paused, dag.is_subdag, dag.is_active, dag.last_scheduler_run, dag.last_pickled, dag.last_expired, dag.scheduler_lock, dag.pickle_id, dag.fileloc, dag.owners, dag.description, dag.default_view, dag.schedule_interval
FROM dag
WHERE dag.is_active AND dag.dag_id = :dag_id_1

SELECT dag.dag_id, dag.is_paused, dag.is_subdag, dag.is_active, dag.last_scheduler_run, dag.last_pickled, dag.last_expired, dag.scheduler_lock, dag.pickle_id, dag.fileloc, dag.owners, dag.description, dag.default_view, dag.schedule_interval
FROM dag
WHERE dag.is_active AND dag.dag_id = :dag_id_1
```

So there is NO fundamental differences between them. But there is minor difference in terms performance (please refer to https://github.com/apache/airflow/pull/4433), and I suspect it comes from the compiling process from python code to sql statement (not sure).

### Suggestion
This change doesn't break or harm anything, even though the improvement it brings is very minor.

@feng-tao Please let me know if you don't think it's necessary to make these two `sqlalchemy filter`-related changes. I'm ok to remove them and only keep another two changes in this PR.

Thanks!

Reply:
It is fine, we could keep those as it doesn’t harm like you said. Thanks for the info.

Reply:
It is fine, we could keep those as it doesn’t harm like you said. Thanks for the info.